cypress/support/commands.js

```diff
@@ -30,17 +46,90 @@ Cypress.Commands.add('kcGetToken', function (user) {
 
 Cypress.Commands.add('kcLogout', function () {
   Cypress.log({ name: 'Logout' });
-  let authBaseUrl = Cypress.env('KEYCLOAK_AUTH_URL');
+  let authBaseUrl = Cypress.env('KEYCLOAK_LOCAL_AUTH_URL');
+  let realm = Cypress.env('KEYCLOAK_REALM');
+  return cy.request({
+    url: `${authBaseUrl}/realms/${realm}/protocol/openid-connect/logout`,
+  });
+});
+
+Cypress.Commands.add('kcLogin', (user) => {
+  Cypress.log({ name: 'Login' });
+  cy.fixture('users/' + user).then((userData) => {
+    let authBaseUrl = Cypress.env('KEYCLOAK_LOCAL_AUTH_URL');
+    let realm = Cypress.env('KEYCLOAK_REALM');
+    let client_id = Cypress.env('KEYCLOAK_API_CLIENTID');
+    let client_secret = Cypress.env('KEYCLOAK_LOCAL_SECRET');
+
+    const base64URLEncode = (str) => {
+      return str.toString('base64').replace(/\+/g, '-').replace(/\//g, '_').replace(/=/g, '');
+    };
+
+    const code_challenge = base64URLEncode(crypto.randomBytes(32));
+
+    cy.request({
```

Comment:
As we discussed, please try direct token url
`{{keycloak_url}}/realms/{{realm}}/protocol/openid-connect/token `
HEADERS:
```json
headers [{"key":"Content-Type","name":"Content-Type","value":"application/x-www-form-urlencoded","type":"text"}]
```
body (query string / ) ->
`client_id `
`grant_type ` :  'password'
`username `
`password`
`client_secret `